### PR TITLE
Shopify - Access Read/Write Shopify Metafields

### LINKED ISSUE #5375: [ACTION] Access Read/Write Shopify Metafields
Shopify

https://shopify.dev/api/storefront/2023-01/objects/Metafield

https://shopify.dev/api/admin-graphql/2023-01/objects/MetafieldAccess

Search by. Read. Write. to shopify metafields would be very useful.

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #5375

### CODE REVIEW

File: components/shopify/actions/common/metafield-actions.mjs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      };
+      const draftOrder = {
+        ...shopify.propDefinitions.draftOrderId,
+        options: async () => {
+          return this.shopify.getDraftOrderOptions();
+        },
+      };
+
+      let props = {};
+      if (ownerResource === "product") {
```
Comment (vellames) [inline components/shopify/actions/common/metafield-actions.mjs:83]: Looks like we always have `ownerId` field in the props. If that's the case, can we start `props` with this property and just remove some of these `ifs`?

**Review by vellames — CHANGES_REQUESTED** [review]

**Review by vellames — APPROVED** [review]
lgtm

**Comment by zecul** [comment]
Hi, I'm not currently able to make this work. I have a list metafield for products called "labels"
image

Currently contains data:
image

I'm updating the data in the format recommended by https://shopify.dev/docs/apps/custom-data/metafields/types#list-types
image

But there is an error returned:
image

**Pull request merged**